x-pack/plugin/ilm/src/main/java/org/elasticsearch/xpack/ilm/IndexLifecycleTransition.java

```diff
@@ -91,7 +92,9 @@ public static void validateTransition(
 
         // Always allow moving to the terminal step or to a step that's present in the cached phase, even if it doesn't exist in the policy
         if (isNewStepCached == false
-            && (stepRegistry.stepExists(policyName, newStepKey) == false && newStepKey.equals(TerminalPolicyStep.KEY) == false)) {
+            && (stepRegistry.stepExists(policyName, newStepKey) == false
+                && newStepKey.equals(TerminalPolicyStep.KEY) == false
+                && newStepKey.equals(new Step.StepKey(lifecycleState.phase(), PhaseCompleteStep.NAME, PhaseCompleteStep.NAME)) == false)) {
```

Comment:
Nit: The `new` here threw me off when I read it the first time. What do you think about adding a helper method in `org.elasticsearch.xpack.core.ilm.PhaseCompleteStep`:

- Either adding  `boolean isFinalStep(StepKey stepKey)` ,
- or to add a final step key constructor:
```
    public static StepKey stepKey(String phase) {
        return new StepKey(phase, NAME, NAME);
    }
```